lib/bootstrap/rubygems.rb

```diff
@@ -43,6 +43,25 @@ def self.reset
       end
     end
 
+    ##
+    # Take a plugin name and get the latest versions available in the gem repository.
+    # @param [String] The plugin name
+    # @return [Hash] The collection of registered versions
+    ##
+    def versions(plugin)
+      require "gems"
+      # This patch is necessary to avoid encoding problems when Net:HTTP return stuff in ASCII format, but
+      # consumer libraries, like the YAML parsers expect them to be in UTF-8. As we're using UTF-8 everywhere
+      # and the usage of versions is minimal in our codebase, the patch is done here. If extended usage of this
+      # is done in the feature, more proper fix should be implemented, including the creation of our own lib for
+      # this tasks.
+      ::Gems::Request.module_eval do
```

Comment:
I understand we cannot move this patch to the `logstash-core` because of the PATH handling, But I think the idea behind @jsvd comment is to externalize this patch outside of where its called. If we ever need to remove it, this patch will be really hidden somewhere and not as obvious as a file in the `logstash/patches` directory, the plugin manager has a few required patch to allow it to work correctly. 

Also having this patch inline within the `#versions` method mean jruby will evaluate the `module_eval` for every method call, in the context of jruby this mean the methods cache will be dropped on every method call.

This kind of behavior from gems/rubygems makes me really sad, I still don't understand were in the chain the utf-8 information of the request isn't made.